### PR TITLE
Automatically append image/gif/video attachment urls to tag content when creating/editing tags

### PR DESCRIPTION
**Pull request by WitherredAway**
This pull request implements a long anticipated feature of being able to send attachments (images/gifs/videos) when creating/editing tags and have their urls be automatically appended to the tag content. Currently the extra hassle of first sending the attachment, copying the link and then pasting into the tag content is needed in order to put attachments into tags. This feature makes it much more intuitive as most (newer) users first try to add attachments this way but to no avail.

This pull request also fixes a bug where there were missing content length checks in edit/forceedit commands, allowing (nitro) users to edit tags to have more than 2000 characters leading to the tag breaking and no longer being shown by the bot due to the bot character limit being 2000, which I assume also raises errors on the bot's end.

Please note that while I have made sure that my code works, I have not tested it on an instance of the bot.

### CODE REVIEW

File: cogs/tags.py [hunk 1]
```diff
@@ -161,10 +164,14 @@ async def list(self, ctx, *, member: discord.Member = None):
     @tag.command()
     @commands.check_any(checks.is_moderator(), commands.has_role("Create Tags"))
     async def create(self, ctx, name, *, content):
-        """Creates a new tag owned by you."""
+        """Creates a new tag owned by you. Attach images/gifs/videos to automatically append their urls to the tag."""
```
Comment (oliver-ni) [inline cogs/tags.py]: Can you reword this as "Attachments will have their URLs appended to the tag."

File: cogs/tags.py [hunk 2]
```diff
@@ -161,10 +164,14 @@ async def list(self, ctx, *, member: discord.Member = None):
...
 
-        if len(content) > 1994:
-            await ctx.send("Tag content must be at most 1994 characters.")
+        for attachment in ctx.message.attachments:
+            if attachment.content_type.split("/")[0] in ("image", "video"):
```
Comment (oliver-ni) [inline cogs/tags.py]: not sure if checking image/video is necessary, attaching other files might be ok too

File: cogs/tags.py [hunk 3]
```diff
@@ -207,10 +221,17 @@ async def edit(self, ctx, name, *, content):
     @tag.command(aliases=("fe",))
     @checks.is_moderator()
     async def forceedit(self, ctx, name, *, content):
-        """Edits a tag by force.
+        """Edits a tag by force. Attach images/gifs/videos to automatically append their urls to the tag.
 
         You must have the Moderator role to use this."""
 
+        for attachment in ctx.message.attachments:
```
Comment (oliver-ni) [inline cogs/tags.py]: this is repeated so maybe pull this into a function with_attachment_urls(ctx, content)

**Review by oliver-ni — CHANGES_REQUESTED** [review]

File: cogs/tags.py (hunk 2 above)
Comment (WitherredAway) [reply cogs/tags.py]: Good point. One concern I have tho is that when you open an attachment url it directly downloads to your device (unless you have it set to prompt each time, which is not the default in some cases) so it may be dangerous in some cases. May just be overthinking, but I'm not sure if other attachments are really necessary. Your call, what do you say?

File: cogs/tags.py (hunk 3 above)
Comment (WitherredAway) [reply cogs/tags.py]: Done! Made some modifications, is this okay?

Comment (WitherredAway) [reply cogs/tags.py]: Will modify the image/video check once the other conversation is resolved

File: cogs/tags.py (hunk 2 above)
Comment (oliver-ni) [reply cogs/tags.py]: i think it's ok. someone could always attach a link to a bad file anyway

Comment (WitherredAway) [reply cogs/tags.py]: Alright 👍

Comment (WitherredAway) [reply cogs/tags.py]: Done!

**Comment by WitherredAway** [comment]
Oop i forgor to make `content` an optional argument

**Comment by WitherredAway** [comment]
I'll open another PR, sorry

**Comment by oliver-ni** [comment]
Np
